server/apps/vocabularies/vocabularies.py

```diff
@@ -50,23 +51,19 @@ def on_fetched(self, doc):
         """
 
         for item in doc[config.ITEMS]:
-            self.__filter_inactive_vocabularies(item)
+            self._filter_inactive_vocabularies(item)
 
     def on_fetched_item(self, doc):
         """
         Overriding to filter out inactive vocabularies and pops out 'is_active' property from the response.
         """
 
-        self.__filter_inactive_vocabularies(doc)
+        self._filter_inactive_vocabularies(doc)
 
-    def __filter_inactive_vocabularies(self, item):
-        active_vocabularies = []
+    def _filter_inactive_vocabularies(self, item):
+        active_vocabularies = [{k: voc[k] for k in voc.keys() if k != 'is_active' and voc.get('is_active', True)}
+                               for voc in item['items']]
```

Comment:
IMO this list should be converted to a generator, because the latter is iterated over again in the very next line, resulting in another list (rendering the initial list kind of redundant).